mtc.go

```diff
@@ -1335,21 +1378,32 @@ func (batch *Batch) ComputeTree(r io.Reader) (*Tree, error) {
 	return &Tree{buf: buf.Bytes(), nLeaves: nLeaves}, nil
 }
 
-// Computes the key of the AbridgedAssertion used in the index.
-func (a *AbridgedAssertion) Key(out []byte) error {
-	buf, err := a.MarshalBinary()
+// Computes the key a BatchEntry for this assertion would have in the index.
+func (a *Assertion) EntryKey(out []byte) error {
+	// We use dummy not_after, as it's ignored in the key.
+	be := a.Abridge(time.Unix(0, 0))
+	return be.Key(out)
+}
+
+// Computes the key of the BatchEntry used in the index.
+//
+// Note that keys are not unique: we leave out the not_after field when
+// computing the key. This allows us to look up a BatchEntry for some
+// assertion that does not contain the not_after field.
+func (be *BatchEntry) Key(out []byte) error {
+	buf, err := be.MarshalBinary()
```

Comment:
Seems like we're doing extra steps here to create a BatchEntry with an empty `notAfter` and omitting it from the hash. Maybe we could add a helper function like below that gets used within `BatchEntry.MarshalBinary` and `Assertion.EntryKey`? And then `BatchEntry.Key` could call `be.Assertion.EntryKey` directly.
```
func (a Assertion) marshal(b *cryptobyte.Builder) {
    ...
}
```

Reply:
The thing is that we can't construct an Assertion from a BatchEntry.